managers/sfrRecord.py

```diff
@@ -354,11 +354,20 @@ def saveEdition(self, edition):
             newEd.sub_title = edition['sub_title'].most_common(1)[0][0]
         newEd.alt_titles = [t[0] for t in edition['alt_titles'].most_common()]
 
-        # Set Publication Data
-        pubYearGroup = re.search(r'([0-9]{4})', str(edition['publication_date']))
-        if pubYearGroup:
-            newEd.publication_date = date(year=int(pubYearGroup.group(1)), month=1, day=1)
+        # Set Publication Date
+        def publicationDateCheck(edition):
```

Comment:
This method is a great start! A couple of requests:
- I think it makes more sense for it to be a method as part of the `SFRRecord` class rather than within this function. I think this is clearer and makes it more testable as well. (can make it a `@staticmethod` for simplicity).
- We should account for UTC time in the date check line, so I think `datetime.utcnow()` instead of `date.today()`. Will this ever make a difference? probably not, but we do use UTC time elsewhere and you never now what could happen